lib/Sema/CSGen.cpp

```diff
@@ -2238,16 +2250,47 @@ namespace {
         Type varType = CS.createTypeVariable(
             CS.getConstraintLocator(locator), TVO_CanBindToNoEscape);
 
+        // When we are supposed to bind pattern variables, create a fresh
+        // type variable and a one-way constraint to assign it to either the
+        // deduced type or the externally-imposed type.
+        Type oneWayVarType;
+        if (bindPatternVarsOneWay) {
+          oneWayVarType = CS.createTypeVariable(
+              CS.getConstraintLocator(locator), TVO_CanBindToNoEscape);
+          if (externalPatternType) {
```

Comment:
I believe that this block could be simplified down to 

```c++
CS.addConstraint(ConstraintKind::OneWayEqual, oneWayVarType, 
                 externalPatternType ? externalPatternType : varType, 
                 locator);
```

Reply:
You're absolutely right, thanks!